app/src/main/java/com/markkolenbrander/capstonenewsapp/views/ArticleViewHolder.kt

```diff
@@ -0,0 +1,27 @@
+package com.markkolenbrander.capstonenewsapp.views
+
+
+import android.view.View
+import androidx.navigation.findNavController
+import androidx.recyclerview.widget.RecyclerView
+import com.markkolenbrander.capstonenewsapp.R
+import com.markkolenbrander.capstonenewsapp.models.Article
+
+class ArticleViewHolder(private val articleView: ArticleView, val detailView: DetailView) : View.OnClickListener ,RecyclerView.ViewHolder(articleView) {
+
+    init {
+       articleView.setOnClickListener(this)
+    }
+
+    fun bindData(article: Article, onDeleteTapped: () -> Unit){
+        articleView.setArticleData(article, onDeleteTapped)
+    }
+
+    fun setDetailData(article: Article){
+        detailView.setArticleDetails(article)
+    }
+
+    override fun onClick(view: View?) {
+        view?.findNavController()?.navigate(R.id.action_articlesListFragment_to_detailFragment)
+    }
```

Comment:
Good start here with the navController! When an article from the list is tapped on, you'll want to bubble that up all the way from the ArticleViewHolder (where you're navigating currently) up to the Adapter then out to the ArticleListFragment.  Think of it terms of roles and responsibilities.  The viewholder should only pass along the click event, the adapter can then pass up the article that was selected, and the ArticleListFragment (because it has direct access to the navcontroller) can trigger a navigation).